ocaml/xapi/xapi_session.ml

```diff
@@ -348,7 +349,9 @@ let slave_local_login ~__context ~psecret =
   Xapi_local_session.create ~__context ~pool:true
 
 (* Emergency mode login, uses local storage *)
-let slave_local_login_with_password ~__context ~uname ~pwd = wipe_params_after_fn [pwd] (fun () ->
+let slave_local_login_with_password ~__context ~uname ~pwd =
+  let pwd = Bytes.of_string pwd in
+  wipe_params_after_fn [pwd] (fun () ->
```

Comment:
Due to the copy won't this leave the original `pwd` unwiped?

Reply:
Until it is GCd away. We cannot really mutate strings or make them mutable with the unsafe conversion, this will have problems with compiler optimisations. The alternative is an invasive change to all the interfaces to accept bytes

Reply:
I have made some tests with the core dumps. It looks like even in normal use, it is GCd fast enough as I could not find it in the core dump at all. I will create a ticket to investigate what to do

Reply:
Thanks, there is a ticket to investigate this, so we should probably not do more in this PR.